case-lib/opt.sh

```diff
@@ -98,21 +96,26 @@ func_opt_parse_option()
 
     define_common_options
 
-    # Asks getopt to validate input and generate _op_temp_script.
-    # Initialize reverse maps _op_short_lst and _opt_long_lst used next.
-    _func_create_tmpbash "$@"
+    local formatted_cmd_opts
+    # short option and long option in getopt format
+    local short_opt_str long_opt_str
+    # the option map used for OPT_VAL overriding
+    local -A short_opt_lst long_opt_lst
+
+    _fill_opt_vars
+    _validate_and_format_cmd_line "$@"
 
-    # FIXME: what is this supposed to do!?
-    eval set -- "$_op_temp_script"
+    # set the contents in formatted_cmd_opts to position arguments $1, $2, ...
+    eval set -- "$formatted_cmd_opts"
```

Comment:
Thanks for the new comment! Can you add a (small) `"$formatted_cmd_opts"` example here?